node/src/block_producer/vrf_evaluator/block_producer_vrf_evaluator_state.rs

```diff
@@ -101,51 +382,555 @@ impl EpochData {
+            ledger: value.ledger.hash,
+            delegator_table: Default::default(),
+            total_currency: value.ledger.total_currency.as_u64(),
+        }
+    }
+}
+
+impl From<ConsensusProofOfStakeDataEpochDataNextValueVersionedValueStableV1> for EpochData {
+    fn from(value: ConsensusProofOfStakeDataEpochDataNextValueVersionedValueStableV1) -> Self {
+        Self {
+            seed: value.seed.into(),
+            ledger: value.ledger.hash,
+            delegator_table: Default::default(),
+            total_currency: value.ledger.total_currency.as_u64(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PendingEvaluation {
+    pub epoch_number: u32,
+    pub epoch_data: EpochData,
+    pub latest_evaluated_slot: u32,
+}
+
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub enum BlockProducerVrfEvaluatorStatus {
-    Idle {
+    /// Initial state
+    Idle { time: redux::Timestamp },
+    /// Waiting for initialization
+    InitialisationPending { time: redux::Timestamp },
+    /// Initialization was a success
+    InitialisationComplete { time: redux::Timestamp },
+    /// Evaluator is ready and able to evaluate an epoch
+    ReadyToEvaluate {
         time: redux::Timestamp,
+        current_epoch_number: u32,
+        is_current_epoch_evaluated: bool,
+        is_next_epoch_evaluated: bool,
     },
-    EpochChanged {
+    /// Waiting for delegator table building
+    EpochDelegatorTablePending {
         time: redux::Timestamp,
+        epoch_number: u32,
+        staking_epoch_ledger_hash: LedgerHash,
     },
-    DataPending {
+    /// Delegator table built successfully
+    EpochDelegatorTableSuccess {
         time: redux::Timestamp,
+        epoch_number: u32,
+        staking_epoch_ledger_hash: LedgerHash,
     },
-    DataSuccess {
+    InitialSlotSelection {
         time: redux::Timestamp,
+        epoch_number: u32,
+        initial_slot: u32,
     },
-    DataFail {
+    /// Epoch evaluation in progress
+    EpochEvaluationPending {
         time: redux::Timestamp,
+        epoch_number: u32,
+        epoch_data: EpochData,
+        latest_evaluated_global_slot: u32,
+        // next_vrf_input: Option<VrfEvaluatorInput>,
     },
-    SlotsRequested {
+    /// A slot was sent for evaluation to the vrf evaluator service
+    SlotEvaluationPending {
         time: redux::Timestamp,
         global_slot: u32,
-        staking_ledger_hash: LedgerHash,
     },
-    SlotsReceived {
+    SlotEvaluationReceived {
         time: redux::Timestamp,
         global_slot: u32,
-        staking_ledger_hash: LedgerHash,
+    },
+    /// The service returned the evaluation succesfully
+    EpochEvaluationSuccess {
+        time: redux::Timestamp,
+        epoch_number: u32,
+    },
+    /// Waiting for the next possible epoch evaluation
+    WaitingForNextEvaluation {
+        time: redux::Timestamp,
+        current_epoch_number: u32,
+        current_best_tip_height: u32,
```

Comment:
In general, I wouldn't duplicate data for best tip or anything that is already stored in state machine. Only times when we do need to duplicate the data, is for example in this case if we need to react to best tip change in a way that we need to know what data from old best tip was and what is the data from new best tip.

In this case, if we need to know those previous values from the best tip in order to decide whether we need to keep "waiting for next evaluation", or to move on. If old values are irrelevant and we can decide just based on the new best tip, then this duplication should be removed. I haven't checked which is the case here.